File: scope106/src/main/java/com/example/scope106/task4/Activity.java
package com.example.scope106.task4;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.example.scope106.R;
import com.example.scope106.databinding.Layout4Binding;

public class Activity extends AppCompatActivity implements View.OnClickListener {
    Layout4Binding binding;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = Layout4Binding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.editTextInput.setOnClickListener(this);
        binding.button.setOnClickListener(this);
    }

    public void startService(View v) {
        String input = binding.editTextInput.getText().toString();

        Intent serviceIntent = new Intent(this, ExampleIntentService.class);
        serviceIntent.putExtra("inputExtra", input);

        ContextCompat.startForegroundService(this, serviceIntent);
    }

    @Override
    public void onClick(View v) {
        if(v.getId() == R.id.button){
            startService(v);
        }
    }
}

File: scope106/src/main/java/com/example/scope106/task4/Activity.kt
package com.example.scope106.task4

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.example.scope106.R
import com.example.scope106.databinding.Layout4Binding

class Activity : AppCompatActivity(), View.OnClickListener {
    var binding: Layout4Binding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = Layout4Binding.inflate(layoutInflater)
        setContentView(binding!!.root)
        binding!!.editTextInput.setOnClickListener(this)
        binding!!.button.setOnClickListener(this)
    }

    fun startService(v: View?) {
        val input = binding!!.editTextInput.text.toString()
        val serviceIntent = Intent(this, ExampleIntentService::class.java)
        serviceIntent.putExtra("inputExtra", input)
        ContextCompat.startForegroundService(this, serviceIntent)
    }

    override fun onClick(v: View) {
        if (v.id == R.id.button) {
            startService(v)
        }
    }
}